pkg/scalers/redis_streams_scaler.go

```diff
@@ -54,15 +54,54 @@ type redisStreamsScaler struct {
 
 type redisStreamsMetadata struct {
 	scaleFactor               scaleFactor
-	targetPendingEntriesCount int64
-	targetStreamLength        int64
-	targetLag                 int64
-	streamName                string
-	consumerGroupName         string
-	databaseIndex             int
-	connectionInfo            redisConnectionInfo
 	triggerIndex              int
-	activationLagCount        int64
+	TargetPendingEntriesCount int64               `keda:"name=pendingEntriesCount,       order=triggerMetadata, optional"`
+	TargetStreamLength        int64               `keda:"name=streamLength,       order=triggerMetadata, optional"`
```

Comment:
nit: I would add `default` here 
```suggestion
	TargetPendingEntriesCount int64               `keda:"name=pendingEntriesCount,       order=triggerMetadata, optional, default=5"`
	TargetStreamLength        int64               `keda:"name=streamLength,       order=triggerMetadata, optional, default=5"`
```

instead of these two places in the `Validate()` method
https://github.com/kedacore/keda/blob/fdfebfa3c303bd7feaea686f83be6aa36970b32a/pkg/scalers/redis_streams_scaler.go#L91-L93

https://github.com/kedacore/keda/blob/fdfebfa3c303bd7feaea686f83be6aa36970b32a/pkg/scalers/redis_streams_scaler.go#L99-L101

other parts of the `Validate()` would probably need to remain. I think that should work.